src/rate-limiting/simple-counting.ts

```diff
@@ -65,24 +60,51 @@ export class SimpleCountingRateLimiter implements RateLimiter {
       this.requestsThisMinute = 0
     }
 
+    return {
+      now,
+      nextSecondInterval,
+      nextMinuteInterval,
+    }
+  }
+
+  msUntilNextExecution(): number {
+    // If the limit is set to infinity, there was no tier limit specified
+    if (this.perSecondLimit === Infinity && this.perMinuteLimit === Infinity) {
+      return 0
+    }
+
+    const { now, nextSecondInterval, nextMinuteInterval } = this.updateIntervals()
+
     const timeToWaitForNextSecond =
-      this.requestsThisSecond + 1 < this.perSecondLimit ? 0 : nextSecondInterval - now
+      this.requestsThisSecond < this.perSecondLimit ? 0 : nextSecondInterval - now
     const timeToWaitForNextMinute =
-      this.requestsThisMinute + 1 < this.perMinuteLimit ? 0 : nextMinuteInterval - now
+      this.requestsThisMinute < this.perMinuteLimit ? 0 : nextMinuteInterval - now
     const timeToWait = Math.max(timeToWaitForNextSecond, timeToWaitForNextMinute)
 
+    return timeToWait
+  }
+
+  async waitForRateLimit(): Promise<void> {
+    const timeToWait = this.msUntilNextExecution()
+
     if (timeToWait === 0) {
-      this.requestsThisSecond++
-      this.requestsThisMinute++
       logger.trace(
-        `Request under limits, counted +1 (S = ${this.requestsThisSecond} | M = ${this.requestsThisMinute})`,
+        `Request under limits, current count: (S = ${this.requestsThisSecond} | M = ${this.requestsThisMinute})`,
```

Comment:
Wouldn't this log be a little redundant with the last one? Since the next one implies the request is under limits?

Reply:
I do like to see current -> update, even if redundant. I know the code will be thread safe cuz JS, but my brain likes to have it because it looks for concurrency problems by default haha

Reply:
I'm good with that 👍